docs/src/query_interface.md

```diff
@@ -0,0 +1,120 @@
+# Query Interface
+
+```@meta
+DocTestSetup =  quote
+    using QuantumSavory
+end
+```
+
+
+## `Tag`
+Tags are identifiers which are used to represent classical information needed for a quantum information. The library allows the construction of custom tags following the format of one of the [`tag_types`](@ref) using the `Tag` constructor. The library implements the following tags for use in the networking protocols:
+
+- `EntanglementCounterpart` 
+    -`remote_node`
+    -`remote_slot`
+    It indicates the current entanglement status with a remote node's slot.
+
+- `EntanglementHistory`
+    - `remote_node`
+    - `remote_slot`
+    - `swap_remote_node`
+    - `swap_remote_slot`
+    - `swapped_local`
+    This tag is used to store the outdated entanglement information after a swap. It helps to direct incoming entanglement update messages to the right node after a swap.
+
+- `EntanglementUpdateX`
+    - `past_local_node`
+    - `past_local_slot`
+    - `past_remote_slot`
+    - `new_remote_node`
+    - `new_remote_slot`
+    - `correction`
+    This tag arrives as a message from a remote node to which the current node was entangled to updat the entanglement information and apply an `X` correction after the remote node performs an entanglement swap.
+
+- `EntanglementUpdateZ`
+    - `past_local_node`
+    - `past_local_slot`
+    - `past_remote_slot`
+    - `new_remote_node`
+    - `new_remote_slot`
+    - `correction`
+    This tag arrives as a message from a remote node to which the current node was entangled to updat the entanglement information and apply a `Z` correction after the remote node performs an entanglement swap.
+
+The tags are constructed using the `Tag` constructor
+#### Tag(tagsymbol::Symbol, tagvariants...)
```

Comment:
This is only one way to construct a tag, and in particular the tags described above are NOT constructed this way.

Reply:
I couldn't get that, isn't this the only way we create tags. Even in ProtocolZoo, we never explicity create instances of the tags described above, its always done through `tag!` which is listed below this part. Is there something I'm missing here?